ui/components/PromptComponent.js

```diff
@@ -60,130 +57,71 @@ const styles = (theme) => ({
     "&:hover" : {
       backgroundColor : "#B32700",
     }
-  },
-  checkboxLabelStyle : {
-    fontSize : "1rem",
-  },
-  checkbox : {
-    color : theme.palette.secondary.focused,
-    "&$checked" : {
-      color : theme.palette.secondary.focused,
-    }
   }
 });
 
-class PromptComponent extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      show : false,
-      title : "",
-      subtitle : "",
-      options : [],
-      isChecked : false,
-      showCheckbox : false
-    }
-    this.promiseInfo = {};
-  }
+function PromptComponent(props) {
+  const [show, setShow] = useState(false);
+  const [title] = useState("");
+  const [subtitle] = useState("");
+  const [options] = useState([]);
 
-  show = async (passed) => {
-    return new Promise((resolve, reject) => {
-      this.promiseInfo = { resolve,
-        reject };
-      this.setState({
-        title : passed.title,
-        subtitle : passed.subtitle,
-        options : passed.options,
-        showCheckbox : !!passed.showCheckbox,
-        show : true
-      });
-    });
+  const hidePrompt = () => {
+    setShow(false);
   };
 
-  hide = () => {
-    this.setState({ show : false });
-  };
+  const { classes } = props;
 
-  handleCheckboxChange = () => {
-    this.setState((prevState) => ({
-      isChecked : !prevState.isChecked,
-    }));
-  };
-
-  getCheckboxState = () => {
-    return this.state.isChecked;
-  };
```

Comment:
Removing these functions will give errors in other modals that use checkbox, I wonder if you have any particular reason to remove them?